tests/publisher_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'rspec'
+require 'simplecov'
+SimpleCov.start
+require_relative '../lib/publisher'
+require_relative '../lib/subscriber'
+
+
+RSpec.describe Subscriber do
+  context '#valid_queue_name?' do
+    let (:valid_queue_name) { "test_123" }
+    
+    it "should validate the url's per aws sqs format" do
+      expect(Publisher.valid_queue_name?(valid_queue_name)).to be_truthy
+    end
+  end
+
+  context '#valid_sleep_period?' do
+    let (:valid_sleep_period) { '7' }
+    it 'subscribes from a queue' do
+      expect(Publisher.valid_sleep_period?(valid_sleep_period)).to be_truthy
```

Comment:
It's not just success values - also test that invalid sleep period results in failure.